tensorflow/c/experimental/filesystem/plugins/hadoop/hadoop_filesystem.cc

```diff
@@ -406,20 +412,24 @@ namespace tf_read_only_memory_region {
 namespace tf_hadoop_filesystem {
 
 void Init(TF_Filesystem* filesystem, TF_Status* status) {
-  filesystem->plugin_filesystem = new LibHDFS(status);
+  filesystem->plugin_filesystem = new HadoopFile({new LibHDFS(status), {}});
   if (TF_GetCode(status) != TF_OK) return;
   TF_SetStatus(status, TF_OK, "");
 }
 
 void Cleanup(TF_Filesystem* filesystem) {
-  auto libhdfs = static_cast<LibHDFS*>(filesystem->plugin_filesystem);
+  auto hadoop_file = static_cast<HadoopFile*>(filesystem->plugin_filesystem);
+  auto libhdfs = hadoop_file->libhdfs;
+  auto connection_cache = &hadoop_file->connection_cache;
   delete libhdfs;
+  delete connection_cache;
```

Comment:
Sorry I've missed this one.
it should be
```cpp
   auto hadoop_file = static_cast<HadoopFile*>(filesystem->plugin_filesystem);
   auto libhdfs = hadoop_file->libhdfs;
   delete libhdfs;
   delete hadoop_file;
```

We need to call `delete` on `hadoop_file` and `libhdfs` since they are the pointers that we allocate in heap and we must clean it to avoid memory leak. On the other hand, we don't call `delete` on the pointer of `connection_cache` because it is in the stack memory ( it will be cleaned up automatically when we `delete hadoop_file;`). Calling `delete` on pointer point to stack memory will cause a seg fault.

Reply:
@vnvo2409 
Can you please review it again?

Reply:
LGTM